### PR TITLE
Pr product funcionallity insert and update product and tests

### PR DESCRIPTION
**Pull request by AleteiaSD**
I have added new functionalities for inserting and updating products. 
Specifically, I implemented new features for inserting products and updating products. 
In addition, I created tests to ensure that these new functionalities are correct and reliable.

During the integration of the new feature branch, I encountered some merge conflicts which I was able to resolve. I also organized the project structure to improve its overall organization and clarity.

I added a new feature called GetProductByGlobalId to implement the update product functionality and added authorize attribute in product controller.

### CODE REVIEW

File: Levi9.POS/Request/ProductRequest/ProductInsertRequest.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿namespace Levi9.POS.WebApi.Request.ProductRequest
+{
+    public class ProductInsertRequest
```
Comment (IvonaSkrinjar) [inline Levi9.POS/Request/ProductRequest/ProductInsertRequest.cs:3]: use DataAnnotations for validation

File: Levi9.POS/Request/ProductRequest/ProductUpdateRequest.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿namespace Levi9.POS.WebApi.Request.ProductRequest
+{
+    public class ProductUpdateRequest
```
Comment (IvonaSkrinjar) [inline Levi9.POS/Request/ProductRequest/ProductUpdateRequest.cs:3]: use DataAnnotations for validation

File: Levi9.POS/Request/ProductRequest/ProductInsertRequest.cs (hunk 1 above)
Comment (AleteiaSD) [reply Levi9.POS/Request/ProductRequest/ProductInsertRequest.cs:4]: I have added DataAnnotations for validation. 
[Required]
public string Name { get; set; }

File: Levi9.POS/Request/ProductRequest/ProductUpdateRequest.cs (hunk 2 above)
Comment (AleteiaSD) [reply Levi9.POS/Request/ProductRequest/ProductUpdateRequest.cs:4]: I have added DataAnnotations for validation. 
[Required]
public string Name { get; set; }

**Review by IvonaSkrinjar — APPROVED** [review]

**Pull request merged**